deploy/objects/Deployer.js

```diff
@@ -1,62 +1,62 @@
+const Package = artifacts.require('Package');
+const AppManager = artifacts.require('AppManager');
+const ContractDirectory = artifacts.require('ContractDirectory');
 const UpgradeabilityProxyFactory = artifacts.require('UpgradeabilityProxyFactory');
 
-const Deployer = {
-  async projectController(projectOwner, projectName, fallbackProvider = 0) {
-    const factory = await UpgradeabilityProxyFactory.new()
-    const registry = await Registry.new({ from: projectOwner })
-    return await ProjectController.new(projectName, registry.address, factory.address, fallbackProvider, { from: projectOwner });
-  },
-
-  async deployAndRegister(projectController, contractKlazz, contractName, version) {
-    const registry = await projectController.registry();
-    const registryManager = new RegistryManager(Registry.at(registry));
-    return await registryManager.deployAndRegister(contractKlazz, contractName, version);
-  },
-
-  async createProxy(projectController, contractClazz, contractName, distribution, version) {
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    return await controllerManager.createProxy(contractClazz, contractName, distribution, version);
-  },
-
-  async createProxyAndCall(controller, owner, contractKlazz, contractName, distribution, version, initArgTypes, initArgs) {
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    return await controllerManager.createProxyAndCall(contractKlazz, contractName, distribution, version, initArgTypes, initArgs);
-  },
-
-  async zepCore(owner, newVersionCost, developerFraction, distribution = 'ZeppelinOS') {
-    const controller = await this.projectController(owner, distribution)
-    const registry = Registry.at(await controller.registry())
-    await this._registerZepCoreDependencies(registry);
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    const kernelStakes = await controllerManager.createProxyAndCall(KernelStakes, 'KernelStakes', distribution, '0', ['address'], [owner]);
-    const kernelRegistry = await controllerManager.createProxyAndCall(KernelRegistry, 'KernelRegistry', distribution, '0', ['address'], [owner]);
-    const zepToken = await controllerManager.createProxyAndCall(ZepToken, 'ZepToken', distribution, '0', ['address'], [owner]);
-    const zepCore = await controllerManager.createProxyAndCall(ZepCore, 'ZepCore', distribution, '0',
-      ['uint256', 'uint256', 'address', 'address', 'address'],
-      [newVersionCost, developerFraction, zepToken.address, kernelRegistry.address, kernelStakes.address]
+class Deployer {
+  constructor(owner) {
+    this.owner = owner;
+  }
+
+  async initAppManager(initialVersion) {
+    const factory = await UpgradeabilityProxyFactory.new({ from: this.owner });
+    this.package = await Package.new({ from: this.owner });
+    this.contractDirectory = await ContractDirectory.new(0, { from: this.owner });
```

Comment:
Two things here:
1. Why are we sending a 0 here? This contract should not receive a param
2. We should not assign this contract directory to `this`, it should be used only within this function context